npc/re/merchants/barters/enchan_illusion_dungeons.yml

```diff
@@ -828,3 +828,83 @@ Body:
           - Index: 3
             Item: Token_Of_Bigstone
             Amount: 150
+  - Name: barter_ill_teddy
+    Items:
+      - Index: 0
+        Item: Counter_Dagger_IL
+        RequiredItems:
+          - Index: 0
+            Item: Counter_Dagger
+          - Index: 1
+            Item: IllusionStone
+            Amount: 95
+          - Index: 2
+            Item: Green_Bearopy
+            Amount: 75
+          - Index: 3
+            Item: Chonchon_Doll_HQ
+            Amount: 50
+      - Index: 1
+        Item: Gate_KeeperDD_IL
+        RequiredItems:
+          - Index: 0
+            Item: Gate_KeeperDD
+          - Index: 1
+            Item: IllusionStone
+            Amount: 95
+          - Index: 2
+            Item: Gothic_China_Doll
+            Amount: 75
+          - Index: 3
+            Item: NasariansSoulstone
+            Amount: 150
+      - Index: 2
+        Item: Survival_Staff_IL
+        RequiredItems:
+          - Index: 0
+            Item: Survival_Rod_
+          - Index: 1
+            Item: Survival_Rod2_
+          - Index: 2
+            Item: IllusionStone
+            Amount: 125
+          - Index: 3
+            Item: Gothic_China_Doll
+            Amount: 100
+          - Index: 4
+            Item: Chonchon_Doll_HQ
+            Amount: 100
+          - Index: 5
+            Item: Clay_Doll
+            Amount: 15
+      - Index: 3
+        Item: Boots_IL
+        RequiredItems:
+          - Index: 0
+            Item: Boots_
+          - Index: 1
+            Item: IllusionStone
+            Amount: 85
+          - Index: 2
+            Item: Green_Bearopy
+            Amount: 50
+          - Index: 3
+            Item: Gothic_China_Doll
+            Amount: 50
+      - Index: 4
+        Item: Headband_Of_Power_IL
+        RequiredItems:
+          - Index: 0
+            Item: Headband_Of_Power
+          - Index: 1
+            Item: IllusionStone
+            Amount: 115
+          - Index: 2
+            Item: NasariansSoulstone
+            Amount: 100
+          - Index: 3
+            Item: Chonchon_Doll_HQ
+            Amount: 50
+          - Index: 4
+            Item: Clay_Doll
+            Amount: 5
```

Comment:
As the limit is 5 items, wouldn't that be correct?
```suggestion
      - Index: 2
        Item: Survival_Staff_IL
        RequiredItems:
          - Index: 0
            Item: Survival_Rod_
          - Index: 1
            Item: IllusionStone
            Amount: 125
          - Index: 2
            Item: Gothic_China_Doll
            Amount: 100
          - Index: 3
            Item: Chonchon_Doll_HQ
            Amount: 100
          - Index: 4
            Item: Clay_Doll
            Amount: 15
      - Index: 2
        Item: Survival_Staff_IL
        RequiredItems:
          - Index: 0
            Item: Survival_Rod2_
          - Index: 1
            Item: IllusionStone
            Amount: 125
          - Index: 2
            Item: Gothic_China_Doll
            Amount: 100
          - Index: 3
            Item: Chonchon_Doll_HQ
            Amount: 100
          - Index: 4
            Item: Clay_Doll
            Amount: 15
      - Index: 4
        Item: Boots_IL
        RequiredItems:
          - Index: 0
            Item: Boots_
          - Index: 1
            Item: IllusionStone
            Amount: 85
          - Index: 2
            Item: Green_Bearopy
            Amount: 50
          - Index: 3
            Item: Gothic_China_Doll
            Amount: 50
      - Index: 5
        Item: Headband_Of_Power_IL
        RequiredItems:
          - Index: 0
            Item: Headband_Of_Power
          - Index: 1
            Item: IllusionStone
            Amount: 115
          - Index: 2
            Item: NasariansSoulstone
            Amount: 100
          - Index: 3
            Item: Chonchon_Doll_HQ
            Amount: 50
          - Index: 4
            Item: Clay_Doll
            Amount: 5
```

Reply:
![Capture d’écran (1946)](https://github.com/rathena/rathena/assets/5231893/5f71f6cb-7227-4dac-9f65-cf772fa4fd30)
5 items are required

Reply:
So the barter is giving an error and is only accepting 5
```
Error: : barter_parseBodyNode: Index 5 is out of bounds. Barters support up to 5 requirements.
Error: : Occurred in file 'npc/re/merchants/barters/enchan_illusion_dungeons.yml' on line 876 and column 12.```

Reply:
Then must be updated the barter shop default to 6?

[src/common/mmo.hpp](https://github.com/rathena/rathena/pull/6508/files#diff-43ec8ae20f372aed4db3e61d12663f58d4bf02be1c7b7a4c5fe822347a5f3149R110)

Reply:
> Then must be updated the barter shop default to 6?
> 
> [src/common/mmo.hpp](https://github.com/rathena/rathena/pull/6508/files#diff-43ec8ae20f372aed4db3e61d12663f58d4bf02be1c7b7a4c5fe822347a5f3149R110)

Correct!
```c
	#define MAX_BARTER_REQUIREMENTS 6
```
![image](https://github.com/rathena/rathena/assets/98105181/99c2fcf8-b66b-4b0e-b965-eaef22880531)